aws/resource_aws_gamelift_fleet.go

```diff
@@ -40,6 +40,11 @@ func resourceAwsGameliftFleet() *schema.Resource {
 				Required: true,
 				ForceNew: true,
 			},
+			"fleet_type": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
```

Comment:
Since the [GameLift API Reference](https://docs.aws.amazon.com/gamelift/latest/apireference/API_CreateFleet.html#gamelift-CreateFleet-request-FleetType) lists the default as `ON_DEMAND`, we should also add that to the resource schema, to prevent a difference for all operators when they upgrade the Terraform AWS Provider. We can also support plan-time validation of this attribute. 👍 

```go
			"fleet_type": {
				Type:     schema.TypeString,
				Optional: true,
				ForceNew: true,
				Default:  gamelift.FleetTypeOnDemand,
				ValidateFunc: validation.StringInSlice([]string{
					gamelift.FleetTypeOnDemand,
					gamelift.FleetTypeSpot,
				}, false),
			},
```